.github/workflows/ci.yml

```diff
@@ -18,13 +18,14 @@ jobs:
       contents: read  # for actions/checkout to fetch code
     name: "${{ matrix.root-pom }} on JDK ${{ matrix.java }} on ${{ matrix.os }}"
     strategy:
+      fail-fast: false
```

Comment:
@cpovirk Huh, that's an interesting take; I think this one is a matter of taste, actually, given what you've shared here. I usually disable `fail-fast` in multi-JVM builds, because maybe there is a failure on a JVM build that isn't caught because it is cancelled before it can surface.

I'm usually staring at the build screen while it runs, so I will cancel runs that are obviously going to fail. `fail-fast` would usually catch these.

On balance, I suppose it helps to have it on while the PR is under development, but I have no opinion after that. Would you rather I remove it or keep it?

Re/ #2193 each of these jobs runs on a separate runner machine, so the `mvn install` step should be running within the scope of a job-specific `.m2` local repository. The `mvn install` step is necessary because Maven has trouble resolving dependencies from projects inside the build. But that shouldn't interfere with the test matrix here, unless I'm missing something.

Reply:
Oops, thanks, I was mixing up `fail-fast` and `continue-on-error`. So my point about "running the main tests if the GWT compile fails" is not actually what we're talking about.

I do see more value in continuing with, e.g., JDK 11 if the JDK 8 build fails because of an annoying problem with old versions of javac :) But then it's also sad to have to wait for JDK 11 to run all the tests before you get a notification about the JDK 8 build failure, and that's how it would often go for our internal workflow. Or maybe not? I'm not actually 100% sure whether Copybara fails the submit (and notifies the user) as soon as it sees a failure or not. (How could I know, given that we have `fail-fast` on? :))

I guess my other fear would be that we're burning resources. I thought maybe I'd once heard that the `google` org's various projects may be competing with each other for resources? If so, then I wouldn't want to spend more on a build that is likely to fail. At this point, that's probably the main thing keeping me from giving it a try to see how Copybara behaves.

(For that matter, we could consider dropping one of the older Java releases to prevent our resource usage from growing on that account. Presumably we'd drop either 11 or 17, since that would keep coverage of the "endpoints." Our internal tests run under 21 nowadays—but obviously with the `SecurityManager` deprecation dialed back by default at the moment :))

I could probably go back and forth on this all day. I'll just leave this out in the internal change for now for the sake of getting the build working under newer JDKs. If you want to queue up `fail-fast` as a separate PR with a link back here, then... we'll probably let it sit forever unless something changes, but if we ever do merge it, your name will get attached :)